superset/utils/core.py

```diff
@@ -1097,11 +1097,13 @@ def merge_extra_form_data(form_data: Dict[str, Any]) -> None:
         {"isExtra": True, **fltr} for fltr in append_adhoc_filters  # type: ignore
     )
     if append_filters:
-        adhoc_filters.extend(
-            simple_filter_to_adhoc({"isExtra": True, **fltr})  # type: ignore
-            for fltr in append_filters
-            if fltr
-        )
+        for key, value in form_data.items():
+            if re.match("adhoc_filter.*", key):
+                value.extend(
+                    simple_filter_to_adhoc({"isExtra": True, **fltr})  # type: ignore
+                    for fltr in append_filters
+                    if fltr
+                )
```

Comment:
```suggestion
        for key in ["adhoc_filters", "adhoc_filters_b"]:
            form_data[key] = form_data.get(key, []) + [
                simple_filter_to_adhoc({"isExtra": True, **fltr})  # type: ignore
                for fltr in append_filters if fltr
            ]
```

I think we can be more explicit about the field names to avoid looping through all form_data and a simple get can avoid db migration? Also, the extended value should be assigned back to `form_data` as L1091 may also see `adhoc_filters` not exists in `form_data`.

Reply:
@ktmud in offline discussions I was the person who proposed using a regex to loop through this, so don't blame @kgabryje for this 😄  My reasoning went something like this:
1) it's fairly uncommon for there to be more than 20 controls. So performance impact of looping through all control keys is negligible
2) While I know that this is put here mostly for the Mixed Timeseries chart, I don't see any reason why someone wouldn't consider having 3 (or more) query sections. Only supporting one additional control called `adhoc_filters_b`, but not, say, `adhoc_filters_c` or `adhoc_filters_2`, feels slightly arbitrary. The logic behind my reasoning here was, "any control that starts with `adhoc_filters` should be considered an adhoc filter control, and extra filters should apply to them all". While both solutions solve this specific problem, I still feel the regex solution is less coupled to the Mixed Timeseries chart than only matching `["adhoc_filters", "adhoc_filters_b"]`

Having said that, this was really just a minor nit and likely irrelevant in the grand scheme of things, so I won't get hung up on this either way 🙂

Reply:
Another argument for using more explicit field names is minimizing unintentional side-effects. While the probability is small, there is no mechanism preventing someone from naming a non adhoc-filter field with `adhoc_filter` in the name. The opposite is also true---one may name an adhoc filters field without it starting with `adhoc_filter`. We've seen it for metrics before, which is why we have the [METRIC_KEYS](https://github.com/apache/superset/blob/760dab9abdb2199b50006e01cb1e9cc0743cf966/superset/viz.py#L104-L113) constant.

Reply:
This is true. However, naming a control that doesn't contain adhoc filters with the prefix "adhoc_filters" feels less likely than than the opposite.

Reply:
True.

Reply:
Thanks for discussion. Do we agree that there's no action/change required here?

Reply:
I still believe in general regex should be used sparingly and explicit is better than implicit. Since you can't stop someone from naming a new filter field things like "series_b_filters", the bug may still come back even with a regex.

In any case, we should need a consistent way of mapping field names to field types because most of the time what you want is "do something special based on filter type", not filed names. One obvious solution is to create an `ADHOC_FILTER_KEYS` constant and put it together with `METRIC_KEYS` in some `constants.py` file, but I agree this by itself probably doesn't make much difference on the grander scheme of things.

Reply:
I agree that we should make all these things as explicit as possible, at a minimum documenting magic behavior like this. While on this subject, the whole notion of `QueryFormData` automatically mapping properties like `metric` and `secondary_metric` to `metrics` in `QueryObject` is really opaque, and should probably both 1) be more thoroughly documented 2) be phased out in coming major versions. But in the short term I propose adding documentation around this functionality to at least explain how this stuff works right now.